atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/thrift/SlicePredicates.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.keyvalue.cassandra.thrift;
+
+import java.nio.ByteBuffer;
+
+import org.apache.cassandra.thrift.SlicePredicate;
+import org.apache.cassandra.thrift.SliceRange;
+import org.immutables.value.Value;
+
+import com.palantir.atlasdb.encoding.PtBytes;
+import com.palantir.atlasdb.keyvalue.cassandra.CassandraKeyValueServices;
+
+public final class SlicePredicates {
+
+    private SlicePredicates() { }
+
+    public static SlicePredicate latestVersionForColumn(byte[] columnName, long maxTimestampExclusive) {
+        return create(
+                Range.singleColumn(columnName, maxTimestampExclusive),
+                Limit.ONE);
+    }
+
+    public static SlicePredicate create(Range range, Limit limit) {
+        SliceRange slice = new SliceRange(
+                range.start(),
+                range.end(),
+                false,
+                limit.value());
+
+        SlicePredicate predicate = new SlicePredicate();
+        predicate.setSlice_range(slice);
+
+        return predicate;
+    }
+
+    @Value.Immutable
+    public interface Range {
+
+        ByteBuffer UNBOUND_START = ByteBuffer.wrap(PtBytes.EMPTY_BYTE_ARRAY);
+        ByteBuffer UNBOUND_END = ByteBuffer.wrap(PtBytes.EMPTY_BYTE_ARRAY);
+
+        Range ALL = Range.of(UNBOUND_START, UNBOUND_END);
+
+        static Range singleColumn(byte[] columnName, long maxTimestampExclusive) {
+            ByteBuffer start = startOfColumn(columnName, maxTimestampExclusive);
+            ByteBuffer end = endOfColumn(columnName);
+            return of(start, end);
+        }
+
+        static ByteBuffer startOfColumn(byte[] columnName, long maxTimestampExclusive) {
```

Comment:
I would rename this to `startColumn`. Every `(columnName, ts)` pair is a Cassandra column.

Reply:
I don't think I would. It's the start of the particular column - `startColumn` would imply to me that `columnName` was the first of many columns.

Reply:
But this is exactly the case with `getRowsColumnRangeForSingleHost`. See CassandraKVSImpl#794